dali/python/nvidia/dali/ops.py

```diff
@@ -554,12 +557,21 @@ def _load_ops():
     _gpu_ops = _gpu_ops.union(set(_b.RegisteredGPUOps()))
     _mixed_ops = _mixed_ops.union(set(_b.RegisteredMixedOps()))
     _cpu_gpu_ops = _cpu_ops.union(_gpu_ops).union(_mixed_ops)
-    for op_name in _cpu_gpu_ops:
-        if not hasattr(sys.modules[__name__], op_name):
-            op_class = python_op_factory(op_name, op_device = "cpu")
-            setattr(sys.modules[__name__], op_name, op_class)
+    ops_module = sys.modules[__name__]
+
+    for op_reg_name in _cpu_gpu_ops:
+        op_full_name = op_reg_name
+        if not op_full_name.startswith('_'):
+            op_full_name = op_full_name.replace('_', '.')
+        *submodule, op_name = op_full_name.split('.')
+        module = _internal.get_submodule(ops_module, submodule)
+        if not hasattr(ops_module, op_full_name):
+            op_class = python_op_factory(op_name, op_reg_name, op_device = "cpu")
+            op_class.__module__ = module.__name__
+            setattr(module, op_name, op_class)
```

Comment:
if you're not placing it in `ops_module`, please change the condition in the `if` above, so that you don't keep setting and wrapping it on reload.

Reply:
done